src/tests/cwrapper/test_cwrapper.c

```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
+#include "cwrapper.h"
+
+void test_cwrapper() {
+    char* s;
+    basic x = basic_new(), y = basic_new(), z = basic_new();
+    symbol_set(x, "x");
+    symbol_set(y, "y");
+    symbol_set(z, "z");
+    
+    s = basic_str(x);
+    printf("Symbol : %s\n", s);
+    basic e = basic_new();
+
+    integer_set_ui(e, 123);
+    s = basic_str(e);
+    printf("Integer : %s\n", s);
+
+    integer_set_ui(e, 456);
+    basic_add(e, e, x);
+    basic_mul(e, e, y);
+    basic_div(e, e, z);
+    printf("Basic : %s\n", basic_str(e));
+    basic_diff(e, e, z);
+    printf("Basic : %s\n", basic_str(e));
+
+    rational_set_ui(e, 100, 47);
+    s = basic_str(e);
+
+    printf("Rational : %s\n", s);
+    printf("Is_a_Symbol %s: %d\n", s, is_a_Symbol(e));
+    printf("Is_a_Rational %s: %d\n", s, is_a_Rational(e));
+    printf("Is_a_Integer %s: %d\n", s, is_a_Integer(e));
```

Comment:
This test should also show how to properly deallocate the memory. As it stands it leaks:
```
certik@redhawk:~/repos/csympy(pr-357)$ valgrind ./src/tests/cwrapper/test_cwrapper 
==25725== Memcheck, a memory error detector
==25725== Copyright (C) 2002-2012, and GNU GPL'd, by Julian Seward et al.
==25725== Using Valgrind-3.8.1 and LibVEX; rerun with -h for copyright info
==25725== Command: ./src/tests/cwrapper/test_cwrapper
==25725== 
Symbol : x
Integer : 123
Basic : (456 + x)*y/z
Basic : -(456 + x)*y/z^2
Rational : 100/47
Is_a_Symbol 100/47: 0
Is_a_Rational 100/47: 1
Is_a_Integer 100/47: 0
==25725== 
==25725== HEAP SUMMARY:
==25725==     in use at exit: 322 bytes in 18 blocks
==25725==   total heap usage: 2,530 allocs, 2,512 frees, 55,724 bytes allocated
==25725== 
==25725== LEAK SUMMARY:
==25725==    definitely lost: 76 bytes in 9 blocks
==25725==    indirectly lost: 246 bytes in 9 blocks
==25725==      possibly lost: 0 bytes in 0 blocks
==25725==    still reachable: 0 bytes in 0 blocks
==25725==         suppressed: 0 bytes in 0 blocks
==25725== Rerun with --leak-check=full to see details of leaked memory
==25725== 
==25725== For counts of detected and suppressed errors, rerun with: -v
==25725== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 6 from 6)
```
It's a little confusing, since in C++ things are deallocated automatically at the right places. We need to manually deallocate the `x`, `y`, `z` symbols, that's clear. I am not 100% sure yet, but I think the `e` expression only needs to be manually deallocated at the very end. For example, in the lines:
```
basic_add(e, e, x);
basic_mul(e, e, y);
basic_div(e, e, z);
```
I think there is no memory leak, as the old expression `e` gets automatically deallocated by C++ inside the `basic_add` function. So only at the very end, we need to deallocate `e`, once. That should do it.



For comparison, if you use valgrind on some other test, you get no leaks:
```
certik@redhawk:~/repos/csympy(pr-357)$ valgrind ./src/tests/basic/test_arit 
==25838== Memcheck, a memory error detector
==25838== Copyright (C) 2002-2012, and GNU GPL'd, by Julian Seward et al.
==25838== Using Valgrind-3.8.1 and LibVEX; rerun with -h for copyright info
==25838== Command: ./src/tests/basic/test_arit
==25838== 
(3/2)*sin(x^2)*cos((3/4)*x)*x/cos(x^2)^2
(3/2)*sin(x^2)*cos((3/4)*x)*x/cos(x^2)^2
(-I*y + x)*(I*y + x)
(-I*y + x)*(I*y + x)
r1: 4
r2: 4
r1: 2
r2: 2
35ms
(w + z + y + x)^10
46ms
number of terms: 286
w*(z + y + x)
w*z + w*x + w*y
(y + x)*(w + z)
w*y + y*z + w*x + x*z
2*x^3 + 2*x^5 + x^2 + 3*x^4 + x^6
x^2 + 2*x^3 + 2*x^5 + 3*x^4 + x^6
x^10 + 2*x^6 + x^2
x^2 + 2*x^6 + x^10
6 + 3*x + 2*y + x*y
6 + x*y + 3*x + 2*y
(w + (w + z + y + x)^2)*(w + z + y + x)^2
4*x*y^3 + y^4 + 4*y^3*z + 4*y*z^3 + z^4 + 12*x*y*z^2 + 4*x*z^3 + 6*w^2*y^2 + 4*w*y^3 + w^4 + x^4 + 4*x^3*y + 6*x^2*y^2 + 4*x^3*z + 6*x^2*z^2 + 12*w^2*x*y + 4*w^3*y + 12*w*x^2*y + 4*w^3*x + 12*w*x*y^2 + 6*w^2*x^2 + 12*w*x^2*z + 12*w^2*y*z + 6*y^2*z^2 + 12*x*y^2*z + 24*w*x*y*z + 4*w^3*z + 12*w*x*z^2 + 12*w*y*z^2 + 4*w*z^3 + 12*w^2*x*z + 6*w^2*z^2 + w*x^2 + 2*w^2*y + 12*x^2*y*z + 2*w*x*y + w^3 + w*y^2 + 2*w*x*z + 2*w*y*z + 4*w*x^3 + w*z^2 + 2*w^2*x + 12*w*y^2*z + 2*w^2*z
10ms
number of terms: 45
-794950134656302081199 + 18988330689345532174800*I + (170384897842227668806160 + 124189267023569140286760*I)*x + (1070006395132744137708900 - 394164239219317088767440*I)*x^2 + (932968275874010419138160 - 3895825928866690524234360*I)*x^3 + (-6988404340183189945649810 - 7535588410733658756109200*I)*x^4 + (-20261671705421920334562768 + 3264389279491708979851032*I)*x^5 + (-13789181346575704878654300 + 30204907418381041842546960*I)*x^6 + (23552046429755146551552720 + 38026705514292667504570680*I)*x^7 + (51144989521911670724567085 + 1712595840361120641326400*I)*x^8 + (29381953248521491520851520 - 41028315045473578363363680*I)*x^9 + (-15338093829390079334615600 - 40586747576398927461290304*I)*x^10 + (-31979670710764393227923520 - 7376058810761038514419680*I)*x^11 + (-16003448831798366397269720 + 15092435518027961390980800*I)*x^12 + (2198647626826273601155520 + 12955420656098394173169120*I)*x^13 + (6418220027431806057632400 + 2865397020090028151964480*I)*x^14 + (2857684148684492876154432 - 1803182138948714915862432*I)*x^15 + (36757437562841466913650 - 1463872202397945728388000*I)*x^16 + (-468934124749482814624800 - 329920367509619821630800*I)*x^17 + (-189467842293333235749000 + 73419306419964967581600*I)*x^18 + (-14133242255825807743200 + 63705777626876324425200*I)*x^19 + (13153337752878940436820 + 13715526624791659668000*I)*x^20 + (4941630430778964093600 - 881242491505751060400*I)*x^21 + (480948022215988899000 - 1101958563746921306400*I)*x^22 + (-141109142617251338400 - 219537463192483503600*I)*x^23 + (-51263209845523090350 - 858036779053272000*I)*x^24 + (-5174179939850141376 + 7486698140478165024*I)*x^25 + (537503012207139600 + 1353369945288761280*I)*x^26 + (204819436022586560 + 43644535485428640*I)*x^27 + (19306159946480680 - 18827472753460800*I)*x^28 + (-568809497106240 - 3042124944711840*I)*x^29 + (-289497261825200 - 123477013792704*I)*x^30 + (-23558947023040 + 15422773148640*I)*x^31 + (-18380219715 + 2196397803600*I)*x^32 + (122189892240 + 82945198440*I)*x^33 + (7811103300 - 3178178640*I)*x^34 + (80276976 - 392830776*I)*x^35 + (-10875410 - 10966800*I)*x^36 + (-454480 + 88920*I)*x^37 + (-3900 + 9360*I)*x^38 + (80 + 120*I)*x^39 + x^40
40ms
number of terms: 40
-794950134656302081199/361188648084531445929920877641340156544317601 - 18988330689345532174800/361188648084531445929920877641340156544317601*I
0ms
==25838== 
==25838== HEAP SUMMARY:
==25838==     in use at exit: 0 bytes in 0 blocks
==25838==   total heap usage: 38,302 allocs, 38,302 frees, 1,182,602 bytes allocated
==25838== 
==25838== All heap blocks were freed -- no leaks are possible
==25838== 
==25838== For counts of detected and suppressed errors, rerun with: -v
==25838== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 6 from 6)
```